File: tests/perf/test-hex-encode.py
import math
import random

def test():
	tmp1 = []
	tmp2 = []

	print('build')
	for i in xrange(1024):
		tmp1.append(chr(int(math.floor(random.random() * 256))))
	tmp1 = ''.join(tmp1)
	for i in xrange(1024):
		tmp2.append(tmp1)
	tmp2 = ''.join(tmp2)

	print(len(tmp2))
	print('run')
	for i in xrange(5000):
		res = tmp2.encode('hex')

test()

File: tests/perf/test-hex-encode.js
if (typeof print !== 'function') { print = console.log; }

function test() {
    var tmp1 = [];
    var tmp2 = [];
    var i, n, buf;

    print('build');
    buf = (ArrayBuffer.allocPlain || Duktape.Buffer)(1024);
    for (i = 0; i < 1024; i++) {
        buf[i] = Math.random() * 256;
    }
    tmp1 = (String.fromBuffer || String)(buf);
    for (i = 0; i < 1024; i++) {
        tmp2.push(tmp1);
    }
    tmp2 = (ArrayBuffer.allocPlain || Duktape.Buffer)(tmp2.join(''));

    print(tmp2.length);
    print('run');
    for (i = 0; i < 5000; i++) {
        // Assigning to 'res' avoids garbage collection of result; this is
        // intentional to avoid mixing string intern performance to the test.
        var res = Duktape.enc('hex', tmp2);
    }
}

try {
    test();
} catch (e) {
    print(e.stack || e);
    throw e;
}
